### ISSUE TITLE
Status bar background drawn below status bar

### ISSUE THREAD

**Issue by filbertteo** [mikepenz/MaterialDrawer#406]
Hi,

Before I begin, I just want to thank you for providing this great library! I have a situation where the status bar background is drawn below the status bar, as shown in the screenshot. It happens when my app returns to portrait orientation after displaying a fullscreen YouTube player view in landscape. I have the android:configChanges option set for my activity in the manifest to prevent my activity from being destroyed and recreated so that the YouTube video will not have to rebuffer. I tested this on 3 devices, and it only happens on Android 4.1.2. The problem doesn't occur on Android 4.4.4 and 5.1.1. I believe the issue is due to the assumption that the status bar doesn't get hidden during fullscreen on pre-sdk 19 devices, or something along those lines. Please help! Thank you!

![screenshot_2015-06-10-12-03-10](https://cloud.githubusercontent.com/assets/12823923/8074872/5e66a666-0f6a-11e5-88d2-cb145ac4edec.png)

**Comment by filbertteo** [comment]
By the way, this doesn't occur with the option `.withTranslucentStatusBar(false)` set. But in that case, my layout doesn't look right on the other 2 devices. But of course, I guess I can always check the SDK version and apply the option if it's < 19. But I believe it will be better if the library can be improved to handle such a situation instead. :)

**Comment by mikepenz** [comment]
@filbertteo i am not sure about the exact code you do, but it is working if you try the sample on an emulator with api level 19. i am sure there seems to be a misconfiguration on your side. 
The library already handles everything. 

Another thing is, that you have to recreate the drawer if the device was rotated. just look at the samples.

**Comment by filbertteo** [comment]
@mikepenz The device with the issue is running Android 4.1.2, i.e. API level 16. It is working fine on API level 19.

Also, is it necessary to recreate the drawer in my case? The activity is never destroyed and recreated, and I can still open and use the drawer as normal. It's just that the status bar background is drawn over my toolbar at the top.

**Comment by mikepenz** [comment]
@filbertteo i give you the advice to let it recreate the activity as the toolbar height is different if you do this.

it will only break after the screen was turned once?

**Comment by filbertteo** [comment]
@mikepenz Yes, it will break only after the device has been rotated. It looks fine at the start. But as I mentioned, I intentionally prevent the activity from being destroyed so that the YouTube video will not need to rebuffer each time the device is rotated.

However, if you feel that this is not a situation that the library is meant to handle, then I will try to implement a workaround on my end. Just thought that it will be good for the library to handle this in case other users also have a similar use case. Thank you!

**Comment by mikepenz** [comment]
@filbertteo i will check what happens if i add the config changes.

**Comment by mikepenz** [comment]
@filbertteo i really can't reproduce this. just tried it again with onConfigChanges. can you send your project? i really want to fix this

**Comment by filbertteo** [comment]
@mikepenz I tried running my project on an emulator, but couldn't get it to work as the YouTube app needs to be installed, which I don't think is possible on an emulator. Do you have a physical device running between API level 14 to 18? If not, you won't be able to test even if I send you my project. I also tested on another physical device running 4.1.1 and the problem occurs as well.

**Comment by filbertteo** [comment]
@mikepenz Anyway, I don't think it's worth your effort trying to hunt this down, as my use case seems to be very specific, since it seems that no one else has encountered this. I can also easily implement a workaround, which I have already done so. But do let me know if you're still keen to investigate why this issue occurs. Cheers :)

**Comment by mikepenz** [comment]
@filbertteo yeah i'm still very interested why it should then draw the statusbar color above the toolbar. I think my HTC Desire is API 19, but will have to take a look. If you send the project (via e-mail) (remove all unnecessary stuff if you are afraid) i will take a look.

**Comment by filbertteo** [comment]
@mikepenz Just sent you an email. Please check. Thanks!

**Comment by mikepenz** [comment]
@filbertteo to which one? To the one in the wiki: mikepenz@gmail.com ?

**Comment by filbertteo** [comment]
@mikepenz Yes, the one in the wiki

**Comment by mikepenz** [comment]
@filbertteo i got no e-mail yet :(

**Comment by filbertteo** [comment]
@mikepenz I think you didn't see my edit for the comment. The email failed to deliver. I just tried sending again, same thing. No reason stated for the failure in the postmaster auto reply.

**Comment by mikepenz** [comment]
@filbertteo then i assume it's because it is an executeable thing. just create a zip. and remove the ".zip" part of the application and send it again. if it's more than 10 mb (not sure if you use gmail but if then 25mb) i upload it to your dropbox or google drive and share it

**Comment by filbertteo** [comment]
@mikepenz Just sent about 3 minutes ago. Haven't gotten a delivery failure notification yet. Let me know if you've received it. Thanks!

**Comment by mikepenz** [comment]
@filbertteo received the mail. will check later ;) still at work

**Comment by filbertteo** [comment]
@mikepenz Sure! Just update me if you manage to find out anything. Thanks!

**Comment by mikepenz** [comment]
@filbertteo the sample you sent should have this issue? i even can't reproduce with this one :(

**Comment by filbertteo** [comment]
@mikepenz Are you sure your device is below API level 19? A suitable device is not capable of displaying a translucent status bar.

**Comment by mikepenz** [comment]
@filbertteo tried it on an emulator with API 17 and lower

**Comment by filbertteo** [comment]
@mikepenz As I said, you need to use a real device because the emulator doesn't have the YouTube app. It's only when the YouTube player becomes fullscreen, then the issue occurs.

**Comment by mikepenz** [comment]
@filbertteo ah yeah true. missed that part while trying :/

**Comment by mikepenz** [comment]
@filbertteo still was not able to reproduce it. but i really had no working device with the same api level. still a problem?

**Comment by filbertteo** [comment]
@mikepenz Actually, due to another incompatibility issue with the YouTube API library (somehow the nagivation bar was "blocking" my YouTube player unless I set it to translucent), I was forced to use the navigation drawer from the Android design support library instead. But even with that library, this issue can occur on those devices if I set `android:fitsSystemWindows="true"`. Apparently different API levels handle that attribute differently, I think. Maybe you can see if that's what's causing the issue when I'm using your library.

**Comment by mikepenz** [comment]
@filbertteo found a solution yet? 

If not can you please test if it still happens with v4.0.0 of the drawer (which is currently in beta testing) 
https://github.com/mikepenz/MaterialDrawer/issues/487

If it still happens the issue will probably find a new place here: https://github.com/mikepenz/Materialize

**Comment by filbertteo** [comment]
@mikepenz Unfortunately, I've already migrated to using the navigation drawer from the Android design support library, so I won't know if your new version fixes the issue. I'm working on other projects for now, but if I have spare time in the future, I'll try to look into it for you. Sorry about that.

**Comment by GitHubMurt** [comment]
I meet this issue on Navigation View using NavDrawer from Google, however it seems to be related.
